api/discovery.capnp

```diff
@@ -0,0 +1,41 @@
+using Go = import "/go.capnp";
+
+@0xfcba4f486a351ac3;
+
+$Go.package("discovery");
+$Go.import("github.com/wetware/ww/internal/api/discovery");
+
+
+interface DiscoveryService {
+    provider @0 (name :Text) -> (provider :Provider);
+    locator @1 (name :Text) -> (locator :Locator);
+}
+
+interface Provider {
+    provide @0 (addrs :Addr) -> ();
+}
+
+interface Locator {
+    findProviders @0 (chan :Sender(Addr)) -> ();
+    
+    using Sender = import "channel.capnp".Sender;
+}
+
+struct Addr {
+    maddrs   @0 :List(Multiaddr);
```

Comment:
Another thought just popped into my head:  this should be an `AddrInfo`, not a list of multiaddrs.  A list of multiaddrs can potentially point to multiple hosts, whereas an `AddrInfo` can only point to a single host.  This is important if we're going to do signature verification, as we don't want to have to check that the signature matches each multiaddr.

Again:  CASM's `boot.capnp` is a good reference.